examples/providers/singleton_thread_locals.py

```diff
@@ -15,7 +15,7 @@ def example(example_object, queue):
 thread_local_object = providers.ThreadLocalSingleton(object)
 
 # Create singleton provider for thread-safe queue:
-queue = providers.Singleton(Queue.Queue)
+queue = providers.ThreadSafeSingleton(queue.Queue)
```

Comment:
Please, rename `queue` to `queue_factory`. As of now, `queue` variable overrides the name of the imported module. It's a bad practice. Also `queue_factory` name will be consistent with `thread_factory` at line 26.